pycontrails/datalib/ecmwf/hres.py

```diff
@@ -635,47 +626,31 @@ def open_metdataset(
             # open cache files as xr.Dataset
             ds = self.open_dataset(disk_cachepaths, **xr_kwargs)
 
-            # TODO: corner case
-            # If any files are already cached, they will not have the version attached
-            if "pycontrails_version" not in ds.attrs:
-                ds.attrs["pycontrails_version"] = pycontrails.__version__
+            ds.attrs.setdefault("pycontrails_version", pycontrails.__version__)
 
         # run the same ECMWF-specific processing on the dataset
         mds = self._process_dataset(ds, **kwargs)
 
-        # convert accumulated radiation values to average instantaneous values
-        # set minimum for all values to 0
-        # !! Note that HRES accumulates from the *start of the forecast*,
-        # so we need to take the diff of each accumulated value
-        # the 0th value is set to the 1st value so each time step has a radiation value !!
-        dt_accumulation = 60 * 60
-
-        for key in [
-            TOAIncidentSolarRadiation.standard_name,
-            TopNetSolarRadiation.standard_name,
-            TopNetThermalRadiation.standard_name,
-        ]:
-            if key in mds.data:
-                if len(mds.data["time"]) < 2:
-                    raise RuntimeError(
-                        f"HRES datasets with data variable {key} must have at least two timesteps"
-                        f" to calculate the average instantaneous value of {key}"
-                    )
-
-                # take the difference between time slices
-                dkey_dt = mds.data[key].diff("time")
-
-                # set difference value back to the data model
-                mds.data[key] = dkey_dt
-
-                # set the 0th value of the data to the 1st difference value
-                # TODO: this assumption may not be universally applicable!
-                mds.data[key][dict(time=0)] = dkey_dt[dict(time=0)]
-
-                rad_accumulated_to_average(mds, key, dt_accumulation)
-
+        self.set_metadata(mds)
         return mds
 
+    @overrides
+    def set_metadata(self, ds: xr.Dataset | MetDataset) -> None:
+        if self.stream == "oper":
+            product = "forecast"
+        elif self.stream == "enfo":
+            product = "ensemble"
+        else:
+            msg = f"Unknown stream type {self.stream}"
+            raise ValueError(msg)
+
+        ds.attrs.update(
+            provider="ECMWF",
+            dataset="HRES",
+            product=product,
+            hres_accumulated=True,
```

Comment:
I didn't notice this on the first pass - are there circumstances where `hres_accumulated` is not True?  We don't have this for ERA5 sources.

Reply:
Based on your comment in CoCiP it looks like this is the case, but just confirming.

Reply:
What about calling this `radiation_accumulated`?

Reply:
aeeacd39055bf8e400cdaf0eab8379ef3d5b2776

Any data coming from our `HRES` interface will have `hres_accumulated=True` because we swap the "step" and the "time" dimension in processing it. But it'd be possible to BYO for which this is not True.